### PR TITLE
Add templatetags `markdownx` with filter `markdown`, and some tests

### PR DESCRIPTION
**Pull request by dyve**

### CODE REVIEW

**Comment by emide** [comment]
+1
Until merged, I add the templatetag on my own projects.

**Comment by Akay7** [comment]
What are we waiting for? I though this is useful PR, or it not compatible with ideology of project?

**Comment by xenatisch** [comment]
@Akay7 For me or @adi- to find some time to make collect everything, make sure they all work, implement some of the things that have been requested, and make a release. 

I really want to do it... and I will at the earliest opportunity (before the end of the year).

**Comment by thaxy** [comment]
Any update?

**Comment by xenatisch** [comment]
@thaxy  Next week... promise.

**Comment by thaxy** [comment]
No stress! Take your time. Awesome work!

**Comment by thaxy** [comment]
I have implemented a custom filter by myself and just want to point out the following as it seems not to be considered in this PR:

* There is no validation if the user saves JavaScript Code in the ``MarkdownxField``. This could lead to XSS vulnerabilities.
* Consider to use ``@stringfilter`` and ``needs_autoescape=True`` as [stated in the docs](https://docs.djangoproject.com/en/2.0/howto/custom-template-tags/#filters-and-auto-escaping).
* Why the use of ``arg=None`` in the function ``markdown``?

**Comment by Akay7** [comment]
@xenatisch on which stage it right now?

**Comment by DoctorJohn** [comment]
The changes proposed by @thaxy are good. But it doesn't look to me that @dyve is still following this discussion. Is there a way for me or someone else to commit those missing changes to this pr?
I tired of having a local "patched" version of django-markdown...

**Comment by dyve** [comment]
Hi, we went for another package when it seemed this would not be made part of django-markdownx.

**Comment by adi-** [comment]
Adding filter to this app is too much risk considering security, as @thaxy wrote above. I think the best what we can do is to leave this request open without implementing it.

**Comment by Akay7** [comment]
@adi- 
Implements web frameworks are too much risky too, but Django exists! Maybe better implements this, but mark in the documentation about that is just for use by an administration, not for drawing any content by any user?

**Comment by adi-** [comment]
Ok, now I think that tonykyriakidis (https://github.com/neutronX/django-markdownx/issues/83#issuecomment-435589618) has quite a good idea that I am taking into consideration. I quite a lot of people are "begging" for it, I guess it could be added to this repo.

**Comment by adi-** [comment]
@dyve, could you update this PR with updated tests?

**Comment by nessalc** [comment]
For this, I use markdownify as a supplemental extension. It uses bleach to sanitize inputs, but you can whitelist tags, styles, attributes, protocols, etc. to your heart's content. Just do `pip install django-markdownify`, put `'markdownify'` in your INSTALLED_APPS, and add

```python
MARKDOWNIFY_WHITELIST_TAGS=['a','h1','h2','h3','h4','h5','h6','p','ol','li','ul',]
MARKDOWNIFY_WHITELIST_ATTRS={'*':['id','alt','class','style','overflow'],
                                  'acronym':['title'],
                                  'a':['href','title'],
                                  'img':['src','width','height','alt'],
                                  'td':['colspan','rowspan'],
                                  'th':['colspan','rowpsan'],
}
MARKDOWNIFY_WHITELIST_STYLES=['color',]
MARKDOWNIFY_WHITELIST_PROTOCOLS=['http','https','mailto',]
MARKDOWNIFY_MARKDOWN_EXTENSIONS=MARKDOWNX_MARKDOWN_EXTENSIONS
```

to `settings.py`. Copy and paste into all your projects.

File: markdownx/templatetags/markdownx.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from django.utils.safestring import mark_safe
+
+from ..utils import markdownify
+
+register = template.Library()
+
+
+@register.filter()
+def markdown(value, arg=None):
+    return mark_safe(markdownify(value))
```
Comment (sitaktif) [inline markdownx/templatetags/markdownx.py]: I would suggest adding:

```python
if not value:
    return ''
```

with a corresponding test below.

**Pull request closed without merging**